### PR TITLE
Navigation: fix navigation when appSubUrl is configured

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
Not sure if this is the best way to solve the issue but this PR solves the bug on main when you've configured Grafana with a subpath.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/core/components/NavBar/NavBarItem.tsx [hunk 1]
```diff
@@ -50,7 +50,9 @@ const NavBarItem = ({
     }
 
     if (!target && url.startsWith('/')) {
-      locationService.push(url);
+      // when config.appSubUrl is configured the locationsService uses appSubUrl on all urls, so we need to strip it before push
+      const path = config.appSubUrl ? url.replace(config.appSubUrl, '') : url;
+      locationService.push(path);
     } else {
```
Comment (torkelo) [inline public/app/core/components/NavBar/NavBarItem.tsx:56]: I am confused by this NavBarItem onNavigate, this is just a normal link, why all this code to navigate to a link?  The code above we already handle in Link component and for normal anchors we have a global intercept as well 

@ashharrison90  ?

Comment (hugohaggmark) [reply public/app/core/components/NavBar/NavBarItem.tsx:56]: @torkelo this is an unfortunate side effect of how react-aria works. They have a very opinionated API where menu items can't contain actionable items so the solution they offer is to use `onAction` here https://github.com/grafana/grafana/blob/7a99b067ef80cd7394bc4ad4326dc4d58a0d8014/public/app/core/components/NavBar/NavBarItemMenuItem.tsx#L42

We did a hacky solution to get around the react-araia limitation before, but we decided to go with their solution with `onAction` in the end:
https://github.com/grafana/grafana/pull/42113/commits/c2fcbfc5ff15ea1ac0e2300ac2360fdfaf55e36f#diff-71fa5f7d566f7c8edcaab336af143a8adca3fa974120cc1978ea48bc2d269d42R61

See the `NOTE: menu items cannot contain interactive content (e.g. buttons, checkboxes, etc.).` here:
https://react-spectrum.adobe.com/react-aria/useMenu.html#complex-menu-items

Comment (hugohaggmark) [reply public/app/core/components/NavBar/NavBarItem.tsx:56]: If the solution offered by this PR is worse than the hack in https://github.com/grafana/grafana/pull/42113/commits/c2fcbfc5ff15ea1ac0e2300ac2360fdfaf55e36f#diff-71fa5f7d566f7c8edcaab336af143a8adca3fa974120cc1978ea48bc2d269d42R61 we can revert to that solution.

Comment (torkelo) [reply public/app/core/components/NavBar/NavBarItem.tsx:56]: Ok, can't say I understand any of that, looks extremely complicated for handling a focus of a link :exploding_head:

File: public/app/core/components/NavBar/NavBarItem.tsx [hunk 2]
```diff
@@ -50,7 +50,9 @@ const NavBarItem = ({
     }
 
     if (!target && url.startsWith('/')) {
-      locationService.push(url);
+      // when config.appSubUrl is configured the locationsService uses appSubUrl on all urls, so we need to strip it before push
+      const path = config.appSubUrl ? url.replace(config.appSubUrl, '') : url;
+      locationService.push(path);
```
Comment (torkelo) [inline public/app/core/components/NavBar/NavBarItem.tsx:55]:
```suggestion
      locationService.push(locationUtil.stripBaseFromUrl(url));
```

Comment (torkelo) [inline public/app/core/components/NavBar/NavBarItem.tsx:55]:
```suggestion
      locationService.push(locationUtil.stripBaseFromUrl(url));
```

Comment (hugohaggmark) [reply public/app/core/components/NavBar/NavBarItem.tsx:55]: Aah, yeah much better. Thank you!

File: public/app/core/components/NavBar/NavBarItem.tsx (hunk 1 above)
Comment (hugohaggmark) [reply public/app/core/components/NavBar/NavBarItem.tsx:56]: It handles much more than focus for a link, it handles all the aria*, keyboard navigation and press events for mobile views. But it's indeed very complex and opinionated.

File: public/app/core/components/NavBar/NavBarItem.tsx (hunk 2 above)
Comment (hugohaggmark) [reply public/app/core/components/NavBar/NavBarItem.tsx]: Commited suggestion, will clean up in a separate commit.

File: public/app/core/components/NavBar/NavBarItem.tsx (hunk 1 above)
Comment (hugohaggmark) [reply public/app/core/components/NavBar/NavBarItem.tsx]: Fixed with suggestion

**Review by ashharrison90 — APPROVED** [review]
lgtm 🙌

**Pull request merged**